filters/openpolicyagent/openpolicyagent.go

```diff
@@ -100,19 +110,23 @@ func WithCleanInterval(interval time.Duration) func(*OpenPolicyAgentRegistry) er
 
 func NewOpenPolicyAgentRegistry(opts ...func(*OpenPolicyAgentRegistry) error) *OpenPolicyAgentRegistry {
 	registry := &OpenPolicyAgentRegistry{
-		reuseDuration:      defaultReuseDuration,
-		cleanInterval:      DefaultCleanIdlePeriod,
-		instances:          make(map[string]*OpenPolicyAgentInstance),
-		lastused:           make(map[*OpenPolicyAgentInstance]time.Time),
-		quit:               make(chan struct{}),
-		maxBodyBytes:       DefaultMaxBodySize,
-		bodyReadBufferSize: defaultBodyBufferSize,
+		reuseDuration:       defaultReuseDuration,
+		cleanInterval:       DefaultCleanIdlePeriod,
+		instances:           make(map[string]*OpenPolicyAgentInstance),
+		lastused:            make(map[*OpenPolicyAgentInstance]time.Time),
+		quit:                make(chan struct{}),
+		maxRequestBodyBytes: DefaultMaxTotalBodySize,
+		bodyReadBufferSize:  defaultBodyBufferSize,
 	}
 
 	for _, opt := range opts {
 		opt(registry)
 	}
 
+	if registry.totalBodySizeSem == nil {
+		registry.totalBodySizeSem = semaphore.NewWeighted(DefaultMaxTotalBodySize)
```

Comment:
this reads a little weird :)
Why the weighted semaphore is created with DefaultMaxTotalBodySize ?
DefaultMaxTotalBodySize sounds like 100MB or 1MB, but that's not the input of the semaphore...

Reply:
The semaphore guards the bytes of the memory allocated for body parsing. This means that Aquire and Release can just use the number of bytes that the filter will consume (content length or max) to make sure the overall.

I am aware that semaphores are typically used for few digit countable resources but I have seen this pattern work well in C++ applications in the past and in this case no assumption on the actual number of concurrent requests need to be made, the max allocated memory is the only fixed value...

Reply:
Yeah makes sense